activetable.js

```diff
@@ -5,12 +5,35 @@ var ActiveTable = (function () {
     NUMERIC = 1;
     STRING = 2;
 
+    // Helper function to determine the number of significant digits in a
+    // floating point literal.  Significant digits are counted from the first
+    // non-zero digit specified, but include trailing zeros.  The string s is
+    // assumed to be a valid floating-point literal and is not validated.
+    function significant_digits(s) {
+        // Regular expression to capture the digits before and after the decimal
+        // point, but not the digits in the exponent.
+        var re = /([0-9]*)\.([0-9]+)|([0-9]+)/;
+        // All specified digits without the decimal point and the exponent.
+        var all_digits = re.exec(s).slice(1).join('');
+        // The number of digits with leading zero digits removed.
+        return all_digits.replace(/^0+/, '').length;
```

Comment:
@smarnach if you're into regex golf, we could merge this replace into the initial regex itself (I'm using my suggested version from comment above):

    /^-?0*([0-9]*)\.?([0-9]*)/

In such case this function becomes one-liner: 

    /^-?0*([0-9]*)\.?([0-9]*)/.exec(s).slice(1).join('').length

It might not be a good idea though :)